src/commands/tail-cloudwatch-logs.js

```diff
@@ -0,0 +1,163 @@
+		global.interval = interval || 1000;
+
+		checkVersion();
+
+		const AWS = getAWSSDK();
+		this.Logs = new AWS.CloudWatchLogs();
+
+		this.log(
+			`looking for CloudWatch log groups with prefix [${namePrefix}] in [${region}]`
+		);
+		const logGroupName = await this.getLogGroupName(namePrefix);
+
+		this.log(`looking for log streams for [${logGroupName}]...`);
+		const logStreamNames = await this.getLogStreamNames(logGroupName);
+
+		this.log(`polling CLoudWatch log group [${logGroupName}]...`);
+		this.log("press <any key> to stop");
+		await this.pollLogGroup(logGroupName, logStreamNames);
+
+		this.exit(0);
+	}
+
+	async getLogGroupName(namePrefix) {
+		const resp = await this.Logs.describeLogGroups({
+			logGroupNamePrefix: namePrefix
+		}).promise();
+
+		if (resp.logGroups.length === 1) {
+			this.log("only 1 match found");
+			return resp.logGroups[0].logGroupName;
+		} else if (_.isEmpty(resp.logGroups)) {
+			this.log(
+				"no matching log groups, please double check the prefix and region and try again"
+			);
+			this.exit(1);
+		} else if (resp.nextToken) {
+			this.log(
+				"too many log groups with matching prefix, please provide the full log group name and try again"
+			);
+			this.exit(1);
+		} else {
+			const logGroupChoices = resp.logGroups.map(x => x.logGroupName);
+			const { logGroupName } = await inquirer.prompt([
+				{
+					type: "list",
+					name: "logGroupName",
+					message: "Which log group are you looking for?",
+					choices: logGroupChoices
+				}
+			]);
+
+			return logGroupName;
+		}
+	}
+
+	async getLogStreamNames(logGroupName) {
+		const resp = await this.Logs.describeLogStreams({
+			logGroupName,
+			descending: true,
+			limit: 50,
+			orderBy: "LastEventTime"
+		}).promise();
+
+		if (_.isEmpty(resp.logStreams)) {
+			this.log("there are no log streams right now, please try again later");
+			this.exit(1);
+		} else {
+			return resp.logStreams.map(x => x.logStreamName);
+		}
+	}
+
+	async pollLogGroup(logGroupName, logStreamNames) {
+		let polling = true;
+		const readline = require("readline");
+		readline.emitKeypressEvents(process.stdin);
+		process.stdin.setRawMode(true);
+		const stdin = process.openStdin();
+		stdin.once("keypress", () => {
+			polling = false;
+			console.log("stopping...");
+		});
+
+		const fetch = async (startTime, endTime, nextToken, acc = []) => {
+			const resp = await this.Logs.filterLogEvents({
+				logGroupName,
+				logStreamNames,
+				interleaved: true,
+				filterPattern: global.filterPattern,
+				startTime,
+				endTime,
+				nextToken
+			}).promise();
+
+			const logMessages = resp.events.map(x => x.message);
+			if (resp.nextToken) {
+				return fetch(startTime, endTime, resp.nextToken, acc.concat(logMessages));
+			} else {
+				return acc.concat(logMessages);
+			}
+		};
+
+		// start from 5s ago
+		let startTime = moment.utc(moment.now()).valueOf() - 5000;
+		let endTime = moment.utc(moment.now()).valueOf();
+		while (polling) {
```

Comment:
Allow the user exit by CTRL-C

Reply:
it already works, it cancels polling when you hit any key